core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/LoadedSignalingInfra.kt

```diff
@@ -86,7 +86,7 @@ interface LoadedSignalInfra {
 interface BlockInfra {
     val blocks: StaticIdxSpace<Block>
 
-    fun getBlockPath(block: BlockId): StaticIdxList<ZonePath>
+    fun getBlockZonePaths(block: BlockId): StaticIdxList<ZonePath>
```

Comment:
After a little discussion, we have a (double) issue to tackle here :
* `blockPath` (not the type but the named concept) already exists, and is a path _inside_ a block, made of...
* `ZonePath`s which represent path _inside_ a zone, made of entry, exit and a sequence of `Node` (actually node-position, but it's redundant with next node and exit).

This contradicts current PR enforcing the idea that a `Object`Path is a _sequence_ of `Object`s.

---

My favorite action on this matter would be to rename:
* `ZonePath` -> `NodePath` (or `SingleZoneNodePath`?)
* `blockPath` -> `nodePathList` (or `singleBlockNodePathList`?)

```suggestion
    fun getNodePathList(block: BlockId): StaticIdxList<NodePath>
```

This is a tedious commit but OK IMO:
* `grep zonepath` and rename in core (not only the type but variables and methods), maybe a bit more?
* any existing `blockPath` in `dev` branch should probably disappear (and new ones may appear in other commits, representing a sequence of blocks).

Reply:
I agree on renaming it, but I'm not sure `NodePath` is better. If anything, the boundaries are detectors. It's still a path though (as in related to a sequence of tracks and other stuff). And it's strongly paired with a single zone. 

I don't have any better suggestion for now, but maybe we could discuss it IRL

Reply:
Soooo... ZonePath, contrarily to all of the other paths, is an infra object, so it's not really the same kind of Path. I doesn't have any kind of offset for example, since it's not linked to a train, it's just a potential path that could be taken through a zone. It's more like a Route or a Block in that sense, and Routes and Blocks aren't considered paths.

If we find a new name, I think we should focus on dropping Path from it... ;)

> blockPath -> nodePathList (or singleBlockNodePathList?)

A block path is a list of blocks (and offsets), which is very much *not* just a list of zonePath. You can have two identical paths through zones, but that don't use the same signaling systems and thus not the same blocks.

Reply:
OK to deal with the rename(s) in another PR and resolve this.
Not sure how to track it: link to this conversation in the parent issue?

Reply:
Tracked here : https://github.com/OpenRailAssociation/osrd/issues/11259